stats/opentelemetry/client_metrics.go

```diff
@@ -87,6 +88,12 @@ func (h *clientStatsHandler) unaryInterceptor(ctx context.Context, method string
 		}
 	}
 
+	// Log an error if one of the options is missing.
+	if (h.options.TraceOptions.TextMapPropagator == nil) != (h.options.TraceOptions.TracerProvider == nil) {
```

Comment:
This check should happen in the constructor for the otel plugin.  Not on every RPC.